utils/eventListeners.ts

```diff
@@ -127,3 +129,88 @@ export function findEventData(
     .filter((event) => `${event.section}.${event.method}` === method)
     .map((event) => event.event.toHuman().data);
 }
+
+export const awaitEvent = async (
+  api: ApiPromise,
+  method: string,
+  blocks: number = 3
+): Promise<void> => {
+  return new Promise(async (resolve, reject) => {
+    let counter = 0;
+    const unsub = await api.rpc.chain.getFinalizedHead(async (head) => {
+      await api.query.system.events((events) => {
+        counter++;
+        testLog
+          .getLog()
+          .info(
+            `await event check for '${method}', attempt ${counter}, head ${head}`
+          );
+        events.forEach(({ phase, event: { data, method, section } }) => {
+          logEvent(phase, data, method, section);
+        });
+        const event = _.find(
+          events,
+          ({ event }) => `${event.section}.${event.method}` === method
+        );
+        if (event) {
+          resolve();
+          unsub();
+          // } else {
+          //   reject(new Error("event not found"));
+        }
+        if (counter === blocks) {
+          reject(`method ${method} not found within blocks limit`);
+        }
+      });
+    });
+  });
+};
+
+export const signSendSuccess = async (
```

Comment:
why not use SignTx from sdk? I think that this could lead to incorrect results because of shuffling, right @mateuszaaa ?